meilisearch-lib/src/index_controller/dump_actor/mod.rs

```diff
@@ -291,21 +292,28 @@ impl DumpJob {
 
         let (sender, receiver) = oneshot::channel();
 
-        self.task_store
-            .register_job(Job::Dump {
+        self.scheduler
+            .write()
+            .await
+            .schedule_job(Job::Dump {
                 ret: sender,
                 path: temp_dump_path.clone(),
             })
             .await;
-        receiver.await??;
-        self.task_store
-            .dump(&temp_dump_path, self.update_file_store.clone())
-            .await?;
+
+        // wait until the job has started performing before finishing the dump process
+        let sender = receiver.await??;
 
         AuthController::dump(&self.db_path, &temp_dump_path)?;
 
+        //TODO(marin): this is not right, the scheduler should dump iteself, not do it here...
+        self.scheduler
+            .read()
+            .await
+            .dump(&temp_dump_path, self.update_file_store.clone())
+            .await?;
+
```

Comment:
```suggestion
        //TODO(marin): this is not right, the scheduler should dump itself, not do it here...
        self.scheduler
            .read()
            .await
            .dump(&temp_dump_path, self.update_file_store.clone())
            .await?;

```

So, should we keep this?